gdx/src/com/badlogic/gdx/utils/Array.java

```diff
@@ -443,6 +443,17 @@ public T random () {
 		return result;
 	}
 
+	@Override
+	public int hashCode () {
+		int h = 1;
+		for (int i = 0, n = size; i < n; i++) {
+			h *= 31;
+			h += items[i] != null ? items[i].hashCode() : 0;

```

Comment:
Could be done in one array lookup. Maybe it would be optimized away, maybe not. We can get a bit more optimizy than usual in collection code.